src/classes/VOL_CTRL_VolunteersBulkEnterHours.cls

```diff
@@ -293,21 +300,28 @@ global virtual with sharing class VOL_CTRL_VolunteersBulkEnterHours {
                     // we couldn't reliably mark dirty records in all scenarios,
                     // so we've decided it is better to always update all records.
                     //if (setVolunteerHoursIdDirty.contains(vh.Id)) 
-                    	listVolunteerHoursUpdate.add(vh);
+                        listVolunteerHoursUpdate.add(vh);
                 } else if (vh.Contact__c != null) {
                     listVolunteerHoursCreate.add(vh);
                 }
             }
+            
+            // Ensure the user can update the object's fields 
+            UTIL_Describe.checkUpdateAccess('Volunteer_Hours__c', new Set<String>{'Contact__c'});            
```

Comment:
since we are using inputField for all the fields on the form the user edits on Hours, the system will be supporting FLS for us.  so we can remove this check.